### PR TITLE
Throughput test for SMT folding in presence of swizzle

### PR DESCRIPTION
**Pull request by shriyak**
SMT folding is achieved through hotplug.To verify this test, run ebizzy workload when SMT is off
and when SMT is on. Observe the records generated for single thread for 60s for one cpu.
Expected is single thread performance is better then multi thread hence it is proved by comparing
the records generated.

Signed-off-by: shriyak <shriyak@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by shriyak** [comment]
Previous PR : https://github.com/avocado-framework-tests/avocado-misc-tests/pull/808

File: cpu/em_smt_folding_test.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        cpu.online(self.cpu)
+        # Disable the idle states
+        process.run("cpupower idle-set -D 0", shell=True)
+        process.system_output("ppc64_cpu --smt=off", shell=True)
+        throughput_off = self.run_ebizzy()
+        process.system_output("ppc64_cpu --smt=on", shell=True)
+        throughput_on = self.run_ebizzy()
+        # Enable the idle states
+        process.run("cpupower idle-set -E 0", shell=True)
+        if (int(throughput_off) > int(throughput_on)):
```
Comment (harish-24) [inline cpu/em_smt_folding_test.py]: Outer braces can be removed.

File: cpu/em_smt_folding_test.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        tarball = self.fetch_asset('http://liquidtelecom.dl.sourceforge.net'
+                                   '/project/ebizzy/ebizzy/0.3'
+                                   '/ebizzy-0.3.tar.gz')
+        archive.extract(tarball, self.srcdir)
+        version = os.path.basename(tarball.split('.tar.')[0])
+        self.sourcedir = os.path.join(self.srcdir, version)
+        patch = self.params.get(
+            'patch', default='Fix-build-issues-with-ebizzy.patch')
+        os.chdir(self.sourcedir)
+        data_dir = os.path.abspath(self.datadir)
```
Comment (harish-24) [inline cpu/em_smt_folding_test.py]: This was as old bug in framework. Please use only `self.datadir` as is.

File: cpu/em_smt_folding_test.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+            deps.extend(['kernel-tools'])
+        for package in deps:
+            if not smg.check_installed(package) and not smg.install(package):
+                self.cancel("%s is needed for the test to be run" % package)
+        tarball = self.fetch_asset('http://liquidtelecom.dl.sourceforge.net'
+                                   '/project/ebizzy/ebizzy/0.3'
+                                   '/ebizzy-0.3.tar.gz')
+        archive.extract(tarball, self.srcdir)
+        version = os.path.basename(tarball.split('.tar.')[0])
+        self.sourcedir = os.path.join(self.srcdir, version)
```
Comment (harish-24) [inline cpu/em_smt_folding_test.py]: Please use `self.workdir` as `srcdir` is getting deprecated.

File: cpu/em_smt_folding_test.py [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+import os
+import platform
+from avocado import Test
+from avocado import main
+from avocado.utils import archive, build
+from avocado.utils import process, cpu, distro
+from avocado.utils.software_manager import SoftwareManager
+
+
+class smt_folding(Test):
```
Comment (harish-24) [inline cpu/em_smt_folding_test.py]: Better to use `SmtFolding` to adhere python standards.

File: cpu/em_smt_folding_test.py [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+        else:
+            self.fail("FAIL : Performance is degraded when SMT=1")
+
+    def run_ebizzy(self):
+        '''
+        Run ebizzy by doing taskset
+        '''
+        output = process.system_output("taskset -c %s ./ebizzy -t1"
+                                       " -S 6 -s 4096" % self.cpu, shell=True)
+        return (output.split('\n', 1)[0]).split(' ')[0]
```
Comment (harish-24) [inline cpu/em_smt_folding_test.py]: Suggestion: Would look better with `output.splitlines()[0]`

**Comment by PraveenPenguin** [comment]
closing in favour of #954

**Pull request closed without merging**